### DISCUSSION TITLE
Nulleinspeisung ohne internet und ohne zusätliche RTCs oder GPS Maus

### DISCUSSION THREAD

**Discussion by VientoFinki** [tbnobody/OpenDTU#1786] [Ideas]
Ich habe das Problem, dass ich eine Nulleinspeisung realisiert habe, die ohne Internet in Rumänien laufen soll. Dort fällt öfter der Strom aus und die Zähler sind always positiv Zähler (man zahlt seinen Strompreis für das was man verbraucht und was man einspeist !! irre!!).
Wenn kein Zeitserver vorhanden ist kann man ja die Zeit manuell per Tastendruck synchronisieren. In der Liveansicht nach dem Booten der openDTU erscheint dann der Hinweis. Das funktioniert auch nur kann man dort nicht nach jedem Stromausfall die Zeit synchronisieren.
Lösungen gibt es aber alle brauchen zusätzliche Geräte.
Meine Idee:
Die Manuelle Zeit Synchronisation gleich beim Hochfahren ausführen lassen wie wenn der Button "Zeit synchronisieren" gedrückt worden wäre. Die Uhrzeit ist dort egal wichtig ist nur dass der MQTT Server läuft und mein Regler auf dem Raspberry die richtigen Werte vom HM-1500 und dem EM3 Shelly einliest das neue Maximum berechnet und auf dem HM-1500 setzt. Was auch alles ohne Internet funktioniert, wenn ich beim Hochstarten mit einer direkten LAN-Verbindung zum Webserver openDTU mit dem Button die Zeit synchronisiere.
Bei meinen bisherigen Software-Änderungen an der openDTU SW konnte ich die Abfrage wegen fehlender Zeitsynchronisation übergehen in dem ich in WebApi_ntp.cpp mal zum Testen dirty die Aufrufe mit getLocalTime immer mit true returnen lasse.

Hat jemand Ideen ob das zum Ziel führt oder wie man das schnell umsetzen könnte ohne GPS Mausoder andere RTCs ?

Ich habe bereits die Lösung selbst gebastelt nur durch eine SW-Änderung der openDTU SW siehe Beitrag in Discussion #82 Was man noch wissen sollte. Aber vielleicht kennt jemand auch eine schicke billige und vor allem  sichere Lösung für den Einsatz in der hintersten Ecke der Welt.

**Comment by MetaChuh** [comment]
> Hat jemand Ideen ob das zum Ziel führt oder wie man das schnell umsetzen könnte ohne GPS Mausoder andere RTCs ?

einfachste lösung:

zwischen usb stromversorgung und dtu eine **usb powerbank** hängen
(billige powerbank, **die beim gleichzeitig laden den ausgang nicht abschaltet**)

poor man's usv, funzt solange die opendtu nicht von selbst rebootet.

**Comment by 3DJupp** [comment]
Bezüglich der Uhrzeit: Ich würde eine RTC/Realtimeclock benutzen. Ein doppelt positiver Zähler ist mir nicht bekannt, aber die Smart-Meter-Verordnung kommt natürlich auch in Rumänien zum Tragen: [https://www.networkedenergy.com/en/success/electrica-romania-smart-energy-transition](https://www.networkedenergy.com/en/success/electrica-romania-smart-energy-transition) 🇪🇺 regelt 😄

**Reply by VientoFinki** [reply]
Danke für den Hinweis. 
Ich kannte das vorher auch nicht. Es war richtig ärgerlich, bis wir es bemerkt haben hat der starke Hoymiles einiges eingespeist, das dann auch noch zum normalen Bezugspreis verrechnet wird. Auch Gespräche mit Elektriker und Wunsch auf einen Zähler mit Rücklaufsperre liefen dort ins Leere (da läuft alles noch anders).  
Wie wäre eine Option in den Einstellungen der DTU und dort mit einer einstellbaren Zeit fortzufahren wie im Shelly EM3 ?

**Comment by spcqike** [comment]
funktioniert die Kommunikation zum Hoymiles denn, wenn die openDTU die falsche Zeit hat?

Wenn dir die tatsächliche Uhrzeit wirklich egal ist, und die Kommunikation mit dem Hoymiles auch mit, sagen wir, 1 Tag Offset klappt, und wenn du sowieso einen raspberry vor Ort hast, warum nimmst du nicht einfach den als Zeitserver?

Installiere die, wenn es nicht bereits drauf ist, den NTP Dienst und lass den Raspberry einen Zeitserver stellen.

Wenn du einen Stromausfall hast, läuft die Uhr im Raspberry natürlich nicht weiter, danach ist er halt raus aus der Echtzeit, aber er startet und hat dann die Uhrzeit, die er zum Stromausfall hatte.

Wenn du es schicker haben willst, hängt eine 5€ RTC in den Raspberry, sodass er beim Starten die halbwegs korrekte Zeit wieder hat.
https://www.berrybase.de/ds3231-real-time-clock-modul-fuer-raspberry-pi
https://forum-raspberrypi.de/forum/thread/50062-ntp-server-mit-ds3231/
